### ISSUE TITLE
[Feature]: Error details as json

### ISSUE THREAD

**Issue by julian-hartl** [BerriAI/litellm#6578]
### The Feature

We would like a better way to handle error responses from the litellm proxy. Currently we get errors like this from the api:
```json
{
  "error":{
    "code":"400",
    "message":"litellm.BadRequestError: AzureException BadRequestError - Error code: 400 - {'error': {'message': \"Sorry! We've encountered an issue with repetitive patterns in your prompt. Please try again with a different prompt.\", 'type': 'invalid_request_error', 'param': 'prompt', 'code': 'invalid_prompt'}}No fallback model group found for original model_group=gpt-4o-mini. Fallbacks=[{'gemini-1.5-pro-preview-0409': ['gemini-1.5-flash', 'gpt-4o']}, {'gemini-1.5-flash': ['gemini-1.5-pro-preview-0409', 'gpt-4o']}, {'gpt-4o': ['gemini-1.5-pro-preview-0409', 'gemini-1.5-flash', 'claude-3-5-sonnet-20240620']}, {'claude-3-5-sonnet-20240620': ['gpt-4o', 'gemini-1.5-pro-preview-0409']}]\nReceived Model Group=gpt-4o-mini\nAvailable Model Group Fallbacks=None\nError doing the fallback: litellm.BadRequestError: AzureException BadRequestError - Error code: 400 - {'error': {'message': \"Sorry! We've encountered an issue with repetitive patterns in your prompt. Please try again with a different prompt.\", 'type': 'invalid_request_error', 'param': 'prompt', 'code': 'invalid_prompt'}}No fallback model group found for original model_group=gpt-4o-mini. Fallbacks=[{'gemini-1.5-pro-preview-0409': ['gemini-1.5-flash', 'gpt-4o']}, {'gemini-1.5-flash': ['gemini-1.5-pro-preview-0409', 'gpt-4o']}, {'gpt-4o': ['gemini-1.5-pro-preview-0409', 'gemini-1.5-flash', 'claude-3-5-sonnet-20240620']}, {'claude-3-5-sonnet-20240620': ['gpt-4o', 'gemini-1.5-pro-preview-0409']}], LiteLLM Max Retries: 1",
    "param":null,
    "type":null
  }
}
```
or sometimes even like this:
```json
{
  "error":{
    "code":"400",
    "message":"litellm.BadRequestError: AzureException BadRequestError - Sorry! We've encountered an issue with repetitive patterns in your prompt. Please try again with a different prompt.No fallback model group found for original model_group=gpt-4o-mini. Fallbacks=[{'gemini-1.5-pro': ['claude-3-5-sonnet-20240620', 'gpt-4o']}, {'gemini-1.5-flash': ['gpt-4o-mini']}, {'gpt-4o': ['gemini-1.5-pro', 'claude-3-5-sonnet-20240620']}, {'claude-3-5-sonnet-20240620': ['gemini-1.5-pro', 'gpt-4o']}]\nReceived Model Group=gpt-4o-mini\nAvailable Model Group Fallbacks=None\nError doing the fallback: litellm.BadRequestError: AzureException BadRequestError - Sorry! We've encountered an issue with repetitive patterns in your prompt. Please try again with a different prompt.No fallback model group found for original model_group=gpt-4o-mini. Fallbacks=[{'gemini-1.5-pro': ['claude-3-5-sonnet-20240620', 'gpt-4o']}, {'gemini-1.5-flash': ['gpt-4o-mini']}, {'gpt-4o': ['gemini-1.5-pro', 'claude-3-5-sonnet-20240620']}, {'claude-3-5-sonnet-20240620': ['gemini-1.5-pro', 'gpt-4o']}], LiteLLM Max Retries: 1",
    "param":null,
    "type":null
  }
}
```

The error messages seem to just be the stringify output of python exceptions.
These are not very ideal, especially the second one, as it is really hard to parse any useful information from them, like the type of error (`invalid_prompt` in the first case) and the actual error message (`Sorry! We've encountered an issue with repetitive patterns in your prompt. Please try again with a different prompt`). 
Instead, I propose to just encode the useful information in json directly. E.g. the error above would look like this:
```json
{
  "error":{
    "code":"400",
    "message":"litellm.BadRequestError: AzureException BadRequestError - Error code: 400 - {'error': {'message': \"Sorry! We've encountered an issue with repetitive patterns in your prompt. Please try again with a different prompt.\", 'type': 'invalid_request_error', 'param': 'prompt', 'code': 'invalid_prompt'}}No fallback model group found for original model_group=gpt-4o-mini. Fallbacks=[{'gemini-1.5-pro-preview-0409': ['gemini-1.5-flash', 'gpt-4o']}, {'gemini-1.5-flash': ['gemini-1.5-pro-preview-0409', 'gpt-4o']}, {'gpt-4o': ['gemini-1.5-pro-preview-0409', 'gemini-1.5-flash', 'claude-3-5-sonnet-20240620']}, {'claude-3-5-sonnet-20240620': ['gpt-4o', 'gemini-1.5-pro-preview-0409']}]\nReceived Model Group=gpt-4o-mini\nAvailable Model Group Fallbacks=None\nError doing the fallback: litellm.BadRequestError: AzureException BadRequestError - Error code: 400 - {'error': {'message': \"Sorry! We've encountered an issue with repetitive patterns in your prompt. Please try again with a different prompt.\", 'type': 'invalid_request_error', 'param': 'prompt', 'code': 'invalid_prompt'}}No fallback model group found for original model_group=gpt-4o-mini. Fallbacks=[{'gemini-1.5-pro-preview-0409': ['gemini-1.5-flash', 'gpt-4o']}, {'gemini-1.5-flash': ['gemini-1.5-pro-preview-0409', 'gpt-4o']}, {'gpt-4o': ['gemini-1.5-pro-preview-0409', 'gemini-1.5-flash', 'claude-3-5-sonnet-20240620']}, {'claude-3-5-sonnet-20240620': ['gpt-4o', 'gemini-1.5-pro-preview-0409']}], LiteLLM Max Retries: 1",
    "param":null,
    "type":null,
    "details":{
      "origin":"AzureException BadRequestError",
      "type":"invalid_request_error",
      "code":"invalid_prompt",
      "param":"prompt",
      "message":"Sorry! We've encountered an issue with repetitive patterns in your prompt. Please try again with a different prompt."
    }
  }
}
```

### Motivation, pitch

We want to implement an error handling layer between litellm and our own services that allows for more granular actions upon errors, such as switching to models with larger context window, when the context window is exceeded, telling our users, when the rate limit is reached, so they can switch models or showing them when their might have been a content filter violation. We also want to be able to display localized versions of these error messages.

**Comment by Keerthi-Iterate** [comment]
This error occurs when the threshold is exceeded: 200 repeated words (e.g., "Hi " * 201, "hello ai\n" * 201) or 100 repeated emojis (e.g., "❤️" * 101, "😉" * 101), with the same 2-token or 3-token sequence.

![image](https://github.com/user-attachments/assets/68acd19c-f6c8-429b-8dc5-cce2f6f85960)

**Comment by julian-hartl** [comment]
> This error occurs when the threshold is exceeded: 200 repeated words (e.g., "Hi " * 201, "hello ai\n" * 201) or 100 repeated emojis (e.g., "❤️" * 101, "😉" * 101), with the same 2-token or 3-token sequence.
> 
> ![image](https://private-user-images.githubusercontent.com/144444273/383491128-68acd19c-f6c8-429b-8dc5-cce2f6f85960.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MDI1NTEsIm5iZiI6MTczMDkwMjI1MSwicGF0aCI6Ii8xNDQ0NDQyNzMvMzgzNDkxMTI4LTY4YWNkMTljLWY2YzgtNDI5Yi04ZGM1LWNjZTJmNmY4NTk2MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQxNDEwNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ODkxZWY1ZWVhZjgxMjhhN2I0OGNmNWE5ZTY4ZTczNzU1NTE4Y2E5MTYwZThjYjNlOTAyMGU0NzM5OTQxNTA1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bF1OfnK8paHMd68JHrfZcn0VteDpliZfdEzcCr7avPc)

Yeah, I know why the error occurs, but you can't really parse that from the error response easily.
